docs/parameters/help_text.md

```diff
@@ -1,12 +1,182 @@
+## Configure help message
+
+Add the following configuration to your configuration files to enable the creation of help messages:
+
+```groovy title="nextflow.config"
+validation {
+    help {
+        enabled: true
+    }
+}
+```
+
+That's it! Every time the pipeline user passes the `--help` and `--helpFull` parameters to the pipeline, the help message will be created!
+
+The help message can be customized with a series of different options. See [help configuration](../configuration/configuration.md#help) docs for a list of all options.
+
+## Help message
+
+Following example shows a snippet of a JSON schema which can be used to perfect visualize the differences between the different help messages. This schema contains one group of parameters called `Input parameters` that contains two parameters: `--input` and `--outdir`. There are also two ungrouped parameters in this schema: `--reference` and `--type`. `--reference` is a nested parameter that contains the `.fasta`, `.fai` and `.aligners` subparameters. `.aligners` also contains two subparameters: `.bwa` and `.bowtie`.
+
+There are three different help messages:
+
+1. Using `--help` will only show the top level parameters (`--input`, `--outdir`, `--reference` and `--type` in the example). The type, description, possible options and defaults of these parameters will also be added to the message if they are present in the JSON schema.
+2. Using `--helpFull` will print all parameters (no matter how deeply nested they are) (`--input`, `--outdir`, `--reference.fasta`, `--reference.fai`, `--reference.aligners.bwa`, `--reference.aligners.bowtie` and `--type` in the example)
+3. `--help` can also be used with a parameter given to it. This will print out a detailed help message of the parameter. This will also show and subparameters present for the parameter.
```

Comment:
```suggestion
3. `--help` can also be used with a parameter given to it. This will print out a detailed help message of the parameter. This will also show the subparameters present for the parameter.
```